### PR TITLE
[Docs] Update pages/concepts/architecture/architecture.md

### LINKED ISSUE #4911: [Docs] Add hyperlinks for each of the architectural components
#### Current State
[This page](https://docs.meshery.io/concepts/architecture) has a couple tables listing architectural components. 

Screen Shot 2022-01-08 at 12 48 58 PM

#### Desired State
Add a hyperlink for each component to its respective docs page.

---

#### [Contributor Guides](https://docs.meshery.io/project/contributing) and Resources
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by verma-kunal**
Signed-off-by: Kunal Verma <vkunal321@gmail.com>

**Description**

This PR fixes #4911 
**Initial PR** for adding hyperlinks for the architectural components mentioned [here](https://docs.meshery.io/concepts/architecture)
Added a few links, but the work in still in progress & undergoing discussions 🙂

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: docs/pages/concepts/architecture/architecture.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Meshery and its components are written using the following languages and technol
 | Meshery Server                                                       | Golang, gRPC, GraphQL, [SMP](https://smp-spec.io)                                 |
 | [Meshery Adapters](/concepts/architecture/adapters)                  | Golang, gRPC, [CloudEvents](https://cloudevents.io/), [SMI](https://smi-spec.io), [OAM](https://oam.dev)  |
 | [Meshery WASM Filters](https://github.com/layer5io/wasm-filters)     | Rust and C++                                                                      |
-| Meshery UI                                                           | ReactJS, NextJS, BillboardJS                                                      |
+| Meshery UI                                                          | ReactJS, NextJS, BillboardJS                                                      |
```
Comment (leecalcote) [inline docs/pages/concepts/architecture/architecture.md:21]: not necessarily a desired change here.

**Comment by l5io** [comment]
🚀 Preview for commit e4166cb05a1164b08383875a67906d5f802859c1 at: https://61dd33d202b3f6ae8aaf8379--meshery-docs-preview.netlify.app

**Review by leecalcote — COMMENTED** [review]
Thanks, @verma-kunal. So far, so good.

File: docs/pages/concepts/architecture/architecture.md (hunk 1 above)
Comment (verma-kunal) [reply docs/pages/concepts/architecture/architecture.md:21]: sure, noted 👍

Comment (leecalcote) [reply docs/pages/concepts/architecture/architecture.md:21]:
```suggestion
| Meshery UI                                                           | ReactJS, NextJS, BillboardJS                                                      |
```

**Review by leecalcote — APPROVED** [review]
Thanks, @verma-kunal 👍 Looks great!

**Pull request merged**